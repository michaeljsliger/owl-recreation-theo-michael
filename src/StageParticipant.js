import React, {Component} from 'react';


class StageParticipant extends Component {
    // what props?
    // ID, Name, Avatar, Onstage, InSession
    // key, name, avatar

    // render template
    render() {
        return (
            <div className="stage-contact" id={this.props.id}>
                <div className="stage-contact-header">
                    {this.props.name}
                </div>
               <img className="stage-contact-image" src={this.props.avatar} alt="huh" />
            </div>
        );
    }
}

export default StageParticipant;