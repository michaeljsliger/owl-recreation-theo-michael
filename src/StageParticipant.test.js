import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import StageParticipant from './StageParticipant';
import renderer from 'react-test-renderer';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('Stage Participant Branch Component Testing', () => {
  // smoke test

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StageParticipant />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  // snapshot test
  it('should render correctly', () => {
    const tree = renderer
          .create(<StageParticipant />)
          .toJSON();
    expect(tree).toMatchSnapshot();
  })
})