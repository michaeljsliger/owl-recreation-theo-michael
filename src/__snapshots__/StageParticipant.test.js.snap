// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage Participant Branch Component Testing should render correctly 1`] = `
<div
  className="stage-piece"
>
  <div
    className="stage-contact"
  >
    <div
      className="stage-contact-header"
    />
    <img
      alt="huh"
      className="stage-contact-image"
    />
  </div>
</div>
`;
